VirtoCommerce.Platform.Web/Scripts/app/dynamicProperties/blades/propertyValue-list.js

```diff
@@ -12,15 +12,17 @@
         var rawProperties = angular.copy(blade.currentEntity.dynamicProperties);
 
         _.each(rawProperties, function (x) {
-            x.values.sort(function (a, b) {
-                return a.value && b.value
-                    ? (a.value.name
-                        ? a.value.name.localeCompare(b.value.name)
-                        : angular.isString(a.value) && angular.isString(b.value)
-                            ? a.value.localeCompare(b.value)
-                            : a.value < b.value ? -1 : a.value > b.value ? 1 : 0)
-                    : -1;
-            });
+            if (x.values) {
+                x.values.sort(function (a, b) {
```

Comment:
not readable code. It is quite difficult to read and understand code like this. I think we should refactor similar code, especially if we are trying to solve some issues which are close to the context.
Why we have here the "magic numbers"?
Why we don't simplify and split some operators here for better readability?
Example:
if (x.values) {
                x.values.sort(function (a, b) {

                  var minusOne = -1; // what is it?
                  var one = 1; // what is it?
                  var zero = 0; // again..
                  var valueA = a.value;
                  var valueB = b.value;
                  var isValueAString = angular.isString(valueA); 
                  var isValueBString = angular.isString(valueB);
                  var nameA = valueA.name;
                  var nameB = valueB.name;
                  var isAEqualsB = nameA.localeCompare(nameB);
                  var isNameAEqualsValueB = valueA.localeCompare(valueB);

                  // here the main logic
                  var someVariable = (nameA ? isAEqualsB : isValueAString && isValueBString
                         ? isNameAEqualsValueB
                         : valueA < valueB ? minusOne : valueA > valueB ? one : zero);
                         
                    return valueA && valueB ? someVariable : minusOne;
                });
            }

It's just an example. I'm sure that here we can apply further decomposition )

Reply:
I agree need to refactor but I added just a condition "if" for safety.. inside it is legacy code. line (15-25)

Reply:
Ok. So be it ) We'll talk about it later